lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -278,125 +315,132 @@ def allocate_address(ec2, domain, reuse_existing_ip_allowed):
         if unassociated_addresses:
             return unassociated_addresses[0], False
 
-    return ec2.allocate_address(domain=domain), True
+    if not domain:
+        domain = 'standard'
```

Comment:
Similar concern as is_aninstance.  This is a strange place to be setting it, and it is not the only place it is set.  That makes it confusing to maintain.